tools/sgapigateway/tools.go

```diff
@@ -0,0 +1,99 @@
+package sgapigateway
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"net/http"
+	"os"
+
+	"go.einride.tech/sage/sg"
+)
+
+//go:embed Dockerfile
+var dockerfile []byte
+
+func EndpointsConfigID(ctx context.Context, serviceName, gcpProject string) string {
+	sg.Logger(ctx).Printf(
+		"retrieving endpoints configID from %s in %s...",
+		serviceName,
+		gcpProject,
+	)
+
+	return sg.Output(sg.Command(
+		ctx,
+		"gcloud",
+		"endpoints",
+		"configs",
+		"list",
+		"--service",
+		serviceName,
+		"--project",
+		gcpProject,
+		"--limit",
+		"1",
+		"--format",
+		"value(id)",
+	))
+}
+
+func DockerImage(ctx context.Context, serviceName, gcpProject, gcpRegion string) string {
+	if err := os.MkdirAll(sg.FromToolsDir("api-gateway"), 0o755); err != nil {
+		panic(err)
+	}
+	configID := EndpointsConfigID(ctx, serviceName, gcpProject)
+	sg.Logger(ctx).Printf(
+		"building image for %s with configID %s...",
+		serviceName,
+		configID,
+	)
+	req, err := http.NewRequestWithContext(
+		ctx,
+		"GET",
+		fmt.Sprintf(
+			"https://servicemanagement.googleapis.com/v1/services/%s/configs/%s?view=FULL",
+			serviceName,
+			configID,
+		),
+		nil,
+	)
+	if err != nil {
+		panic(err)
+	}
+	token := sg.Output(sg.Command(ctx, "gcloud", "auth", "print-access-token"))
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		_ = resp.Body.Close()
+	}()
+	f, err := os.Create(sg.FromToolsDir("api-gateway/service.json"))
+	if err != nil {
+		panic(err)
+	}
+	if _, err := f.ReadFrom(resp.Body); err != nil {
+		panic(err)
+	}
+	defer func() {
+		_ = f.Close()
+	}()
+	tag := fmt.Sprintf(
+		"%s-docker.pkg.dev/%s/docker/endpoints-runtime-serverless:%s-%s",
+		gcpRegion,
+		gcpProject,
+		serviceName,
+		configID,
+	)
+	if err := os.WriteFile(sg.FromToolsDir("api-gateway/Dockerfile"), dockerfile, 0o600); err != nil {
+		panic(err)
+	}
+	cmd := sg.Command(ctx, "docker", "build", "-t", tag, ".")
```

Comment:
Would we benefit from using a standalone tool here instead of docker which we can download such as [buildah](https://buildah.io/) or Kaniko?

Reply:
Coolio, lets but it on the backlog to add a tool going forward :)